App/Sagas/ThreadsSagas.ts

```diff
@@ -0,0 +1,157 @@
+  }
+}
+
+async function getDefaultThread (): Promise<TextileTypes.Thread | undefined> {
+  const threads = await TextileNode.threads()
+  var defaultThread = threads.items.find(thread => thread.name === 'default')
+  return defaultThread
+}
+
+export function * pendingInvitesTask () {
+  // Process any pending external invites created while user wasn't logged in
+  const threads = yield select(ThreadsSelectors.threads)
+  if (threads.pendingInviteLink) {
+    yield call(DeepLink.route, threads.pendingInviteLink, NavigationService)
+    yield put(ThreadsActions.removeExternalInviteLink())
+  }
+}
+
+export function * refreshThreads () {
+  try {
+    const threads: TextileTypes.Threads = yield call(TextileNode.threads)
+    for (const thread of threads.items) {
+      yield put(TextileNodeActions.getPhotoHashesRequest(thread.id))
+    }
+    yield put(ThreadsActions.refreshThreadsSuccess(threads))
+  } catch (error) {
+    yield put(ThreadsActions.refreshThreadsError(error))
+  }
+}
+
+export function * addThread (action: ActionType<typeof ThreadsActions.addThreadRequest>) {
+  const { name, mnemonic } = action.payload
+  try {
+    const thread: TextileTypes.Thread = yield call(TextileNode.addThread, name, mnemonic)
+    yield put(ThreadsActions.addThreadSuccess(thread))
+    yield put(TextileNodeActions.getPhotoHashesRequest(thread.id))
+  } catch (error) {
+    yield put(ThreadsActions.addThreadError(error))
+  }
+}
+
+
+export function * removeThread (action: ActionType<typeof ThreadsActions.removeThreadRequest>) {
+  const { id } = action.payload
+  try {
+    // TODO: something with this blockId
+    const blockId: string = yield call(TextileNode.removeThread, id)
+    yield put(ThreadsActions.removeThreadSuccess(id))
+    yield call(NavigationService.goBack)
+  } catch (error) {
+    yield put(ThreadsActions.removeThreadError(error))
+  }
+}
+
+function bindCallbackToPromise() {
+  let _resolve
+  const promise = () => {
+    return new Promise((resolve) => {
+      _resolve = resolve
+    })
+  }
+  const cb = (args) => _resolve(args)
+
+  return {
+    promise
+    , cb
+  }
+}
+
+async function displayAlert(message: string) {
+  const alert = new Promise<void>((resolve, reject) => {
+    Alert.alert(
+      'Accept Invite',
+      message,
+      [
+        {
+          text: 'Accept',
+          onPress: resolve
+        },
+        {
+          text: 'Ignore',
+          style: 'cancel',
+          onPress: reject
+        }
+      ],
+      {cancelable: false}
+    )
+  })
+  await alert
+}
+export function * reviewThreadInvite (action: ActionType<typeof ThreadsActions.reviewThreadInvite>) {
+  const {notification} = action.payload
+  const payload = NotificationsServices.toPayload(notification)
+  if (!payload) return
+
+  try {
+    yield call(displayAlert, payload.message)
+    yield put(ThreadsActions.acceptInviteRequest(notification.id, notification.category))
+  } catch (error) {
```

Comment:
Need to figure out the right way to do this @asutula, display an alert from the saga so that I can trigger it from anyplace in the app. similar to the comment you left on the last pr, but no idea if the same actually. any idea?

Reply:
Yea should be possible. What's missing at this point? Looks like you're calling `displayAlert` a few lines above. Is that not it?

Reply:
Seems to work, just didn't know if it was a good way.